### DISCUSSION TITLE
Option to omit field from model

### DISCUSSION THREAD

**Discussion by BatmanAoD** [99designs/gqlgen#2040] [Q&A]
Is it possible to omit a field from the generated model of a type in the schema? 

### Minimal graphql.schema and models to reproduce

```graphql
type Bar {
    fooId: String @deprecated(reason: "change in capitalization")
    fooID: String
}
```

I would like to be able to generate the following Go type:

```go
type Bar struct {
    FooID string `json:"fooID"`
}
```

### This may be an [XY problem](https://mywiki.wooledge.org/XyProblem)

The original need I have is to change the capitalization on a field in the schema in a way that causes a collision in the generated Go field names. To make a breaking change like this, I keep the old field but deprecate it while introducing the new field. So, in the example above, to rename `fooId` to `fooID`, I would deprecate `fooId` and introduce `fooID` in a single commit, then later remove `fooId` once all clients have switched over.

But in Go, both `fooId` and `fooID` are given the field name `FooID`. I can explicitly rename one of the fields in the config, so that `FooID` and `FooId` both exist (giving a lint warning about the use of `Id` unless I do something weirder like `DeprecatedFooID`). But in some cases, there's not actually any need for the deserialized version to have the deprecated field, because the value can (for whatever reason) be ignored in the code that uses the models. Hence, in my case, I'd like to simply skip generating that field.

### versions

Since this is a feature request rather than a bug report, I haven't actually tried this because I don't see anything in the [config documentation](https://gqlgen.com/config/) indicating that it's possible.

**Comment by nbpeth** [comment]
I can't seem to comment - doesn't seem to be much of a discussion if I can't, so I'm putting this in the answer block:


Any traction n this? My use-case is tangentially related; I am trying to convert a schema where types may have nested queries or mutations and I definitely do not want them added to the operation.  e.g.

```
type X {
	name: String
	description: String
	someNestedMutation(data: String): String
}
```

Where I do not want `someNestedMutation` as part of the generated operation as it may produce unwanted side effects

**Comment by m4rcLs** [comment]
I have kind of the same problem but additonally I also have to work around deprecated query arguments. This is an example schema which generates broken code.
```gql
type Query {
    ancestors(
        node_id: ID @deprecated(reason: "Use camel case instead")
        nodeId: ID
    ): [Ancestor!]!
}
type NodeType {
    id: ID!
    name: String!
}
type Ancestor {
    id: ID!
    name: String!
    node_type: NodeType! @deprecated(reason: "Use camel case instead")
    nodeType: NodeType!
}
```

For model fields I used the FieldHook to prepend `Deprecated` to every deprecated field.
This works but also necessitates that both fields are set and checked everytime (something I can life with)
To allow multiple json tags per struct field one would need to use something like https://github.com/json-iterator/go instead of the default `encoding/json`.

The main problem I have though is that the generated server code has invalid function definitions which duplicates arguments.
Example from `graph/generated/server.go`:
```go
type ComplexityRoot struct {
	// ...
	Query struct {
		Ancestors func(childComplexity int, nodeID *string, nodeID *string) int
	}
}

type QueryResolver interface {
	Ancestors(ctx context.Context, nodeID *string, nodeID *string) ([]*model.Ancestor, error)
}
```

As far as I know there are no hooks for arguments at the moment which is why I need to always adjust the generated code manually.
Is there any other way around this?
